src/client/cli/cmd/launch.cpp

```diff
@@ -262,7 +262,11 @@ mp::ParseCode cmd::Launch::parse_args(mp::ArgParser* parser)
         if (remote_image_name.startsWith("file://"))
         {
             // Convert to absolute because the daemon doesn't know where the client is being executed.
-            auto file_info = QFileInfo(remote_image_name.remove(0, 7));
+            if (remote_image_name.remove(0, 7)[0] == '~')
+            {
+                remote_image_name.replace(0, 1, QDir::homePath());
+            }
+            auto file_info = QFileInfo(remote_image_name);
```

Comment:
Can't we use an existing utility to parse `file://` urls? There will be other things that this does not cover and we shouldn't have to implement all of them.

Reply:
Hey @ricab, thanks for reviewing this! Right, it would be cleaner to do that. I investigated on this, maybe `POCO::Uri` can help us with `file://`. But, on the other hand, `~` is usually expanded directly by the shell, and having `file://` in front of it makes the shell think it is not used as a home folder. Maybe that's the reason why Qt and `std::filesystem` do not implement `~` expansion.

So moving to an external utility to handle `file://` would not solve our original problem, the `~` expansion. I think making special code to handle this only case (i.e., that the character after `file://` is a `~`) is the way to go. But maybe there is something I am missing. WDYT?

Reply:
That said, let me point out that `QUrl` can handle `file://` and recognize if a given string is a local file or not. See [here](https://doc.qt.io/qt-6/qurl.html#isLocalFile).

Reply:
Ohhh, no. `QUrl` expects `file:` instead of `file://`: we are on the same situation than before.

Reply:
Hmm, honestly I think we should delegate on something that behaves (mostly) correctly and not try to support anything further. If Qt and Poco do not expand `~`, do we really need to? We would then need to [go dig](https://url.spec.whatwg.org/) to find exactly what is and isn't a valid URL. That is a complex standard and we should not be in the business of handling that type of detail.

Reply:
I agree that can leave this to the shell and if the shell doesn't expand it, then it's not our problem. For example, on Firefox, writing in the address bar `file://~/multipass-coverage/index.html ` does not expand `~`. So leaving to the shell would be correct.

OTOH, I think every Linux user (and I am speaking about Linux because for the time being launching blueprints from file is only supported in Linux) expects `~` to be expanded as the home folder. So deciding whether or not to implement this expansion is a matter of being friendly to users or being conservative. I don't have a strong opinion on this, honestly ;)

Reply:
Oh, I answered your first comment without reading the second one (it just arrived). Yeah, I'm seeing there are some options to read better the file://, but the point is to decide whether we want to expand `~`. If not, we can leave everything as it is now and forget about this PR.

Reply:
@ricab, something more about this: if we choose to expand `~`, we'll have to expand also `~user`, and I think that's not what we are supposed to do. So I think we shouldn't expand `~`, but leave it to the shell. And, considering the second and third issues the user reported are not bugs, I vote for closing this PR. WDYT?

Reply:
OK, I am fine with that. I am not sure how `~` and `~user` are supposed to be handled in the URL standard to be honest. I wouldn't mind seeing us delegating URL parsing, but we don't have to do it here or now.